timeline-chart/src/layer/time-graph-navigator.ts

```diff
@@ -104,4 +113,56 @@ export class TimeGraphNavigatorHandle extends TimeGraphComponent<null> {
             color: 0x777769
         })
     }
+}
+
+export class TimeGraphNavigatorBackground extends TimeGraphComponent<null> {
+
+    protected snapEvent: CustomEvent;
+    protected snapEventString: string;
+
+    constructor(protected unitController: TimeGraphUnitController, protected stateController: TimeGraphStateController) {
+        super("navigator_background")
+        this.addEvent("mousedown", event => {
+            let x = event.data.getLocalPosition(this._displayObject).x;
+            let middle = BIMath.round((x / this.stateController.canvasDisplayWidth) * this._displayObject.width);
+            // We have horizontal offset at point of click, now we need coord for start of handler.
+            let hVL = this.unitController.viewRangeLength / BigInt(2);
+            let start0 = middle - hVL;
+            let max = this.unitController.absoluteRange - this.unitController.viewRangeLength;
+            let min = BigInt(0);
+            // Clamp it.
+            const start = BIMath.clamp(start0, min, max)
+            this.unitController.viewRange = {
+              start,
+              end: start + this.unitController.viewRangeLength,
+            }
+            // Set snapped state
+            this.toggleSnappedState(true);
+        }, this._displayObject);
+        // Custom event let's handler know 'mouseup' triggers.
+        this.snapEvent = new CustomEvent(this.snapEventString = 'snap-x-end')
+        const endSnap= () => {
+          this.toggleSnappedState(false);
+          document.dispatchEvent(this.snapEvent)
+        }
+        this.addEvent('mouseup', endSnap, this._displayObject)
+        this.addEvent('mouseupoutside', endSnap, this._displayObject)
+    }
+
+    protected toggleSnappedState = (bool: boolean) => {
+      this.stateController.snapped = bool;
+    }
+
+    render(): void {
+      this.rect({
+        height: 20,
+        position: {
+            x: 0,
+            y: 0
+        },
+        width: Number(this.unitController.viewRangeLength),
+        color: 0xe100ff,
```

Comment:
Just curious, color value is irrelevant since opacity is 0, right?

Reply:
Yes, can't see color w/ opacity = 0.  This was to visually see the background width.  Color has been removed.